templates/_worker.js/routes-matcher.ts

```diff
@@ -364,19 +369,26 @@ export class RoutesMatcher {
 		route: VercelSource,
 		phase: VercelPhase
 	): VercelSource {
+		if (!this.locales || phase !== 'miss') {
+			return route;
+		}
+
+		if (/^\//.test(route.src) && route.src.slice(1) in this.locales) {
+			return { ...route, src: `^${route.src}$` };
+		}
+
+		const trailingSlashRegex = /^\^\/\/\?\(\?:([a-zA-Z|]+)\)\/\(\.\*\)$/;
```

Comment:
A regex of a regex is kind of difficult to follow I think 😓, what about refactoring this in a more readable (although more verbose) way?

Something like this for example?
![Screenshot 2023-06-15 at 11 24 09](https://github.com/cloudflare/next-on-pages/assets/61631103/40fc8607-056b-478f-af93-fa830d612b90)

If you prefer, it can be a method of the class, it can also still use regexes, etc... but I feel like at the very least separating it and adding comments explaining what's going on would actually help making it more understandable and maintainable

what do you think? 🙂

Reply:
sure, i mean, i dont really mind. if you dont like the regex of regex i have no issues changing it to be more like that

Reply:
I do like regexes so if you strongly prefer the regex of regex I am fine with it, although as I mentioned I feel like we need to add comments and make things clearer as the code itself in its current state seems to me too difficult/inconvenient to grasp.

I'm also suggesting to have this logic into a separate function/method so that the if guard can be a clear oneliner, I think it also helps in making things clearer, but if you dislike that I'm fine of having it all in the same method.

tldr; I just think we should try to make this a bit clearer if possible 🙂